hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobProxy.java

```diff
@@ -63,6 +67,15 @@ public JobStatus getStatus() {
         }
     }
 
+    @Nonnull @Override
+    public JobMetrics getMetrics() {
+        try {
+            return JobMetricsImpl.of(this.<Map<String, Long>>invokeOp(new GetJobMetricsOperation(getId())).get());
```

Comment:
JobMetrics instance should be returned inside the operation directly

Reply:
Ok, next version will do so.